straxen/plugins/event_processing.py

```diff
@@ -341,6 +357,31 @@ def fill_result_i(self, event, peaks):
                                                    field_names,
                                                    peak_properties_to_save)
 
+    @staticmethod
+    @numba.njit
+    def set_alt_s2_properties(largest_s1s, s2_idx, largest_s2s, drift_time_max):
+        """Require alt_s2 happens between main S1 and maximum drift time"""
+        if len(largest_s1s) > 0 and len(largest_s2s) > 1:
+            # If there is a valid s1-s2 pair and has a second s2, then check alt s2 validity
+            alt_s2_idx, largest_alt_s2s = s2_idx[1:], largest_s2s[1:]
+            alt_s2_after_s1 = largest_alt_s2s['center_time'] > largest_s1s[0]['center_time']
+            alt_s2_before_max_drift_time = (largest_alt_s2s['center_time'] - largest_s1s[0]['center_time']) < 1.01 * drift_time_max
+            mask = alt_s2_after_s1 & alt_s2_before_max_drift_time
+
+            if (~mask).all():
+                # If no other s2 survives, drop the alt s2 field
+                s2_idx = s2_idx[0:1]
+                largest_s2s = largest_s2s[0:1]
+            else:
+                # Take the first valid largest other s2 as alt s2
+                new_alt_s2_index = np.arange(0, len(largest_alt_s2s))[mask][0] + 1
+                s2_idx[1], s2_idx[new_alt_s2_index] = s2_idx[new_alt_s2_index], s2_idx[1]
+                largest_s2s[1], largest_s2s[new_alt_s2_index] = largest_s2s[new_alt_s2_index], largest_s2s[1]
```

Comment:
Sorry, but here I do not get why you have to move the the index/S2 which is at position 1 to the position of your new alt S2. I mean in the end you only keep the first two values in the array anyhow,